### DISCUSSION TITLE
[Question] Configure YARP to run multiple web servers under one IP address

### DISCUSSION THREAD

**Discussion by ibrahim-cmsb** [dotnet/yarp#981] [Q&A] [answered]
I was recommended to use YARP by a friend. Thanks to all contributors for their amazing work.

I want to use YARP to combine multiple web servers to run as one site. At the moment, all the servers share the same IP address but different ports. For example, a sub-directory with the name `/login` is to redirect to my authentication web server which has its own pages and includes its own resources such as css and js files. The table below gives an example of my current setup.

```
+-----------------+---------------+-------------------+
|     Server      | Sub-directory |     Redirect      |
+-----------------+---------------+-------------------+
| Main            | /             | 192.168.0.10:5000 |
| Authentication  | /login        | 192.168.0.10:5001 |
| User Management | /users        | 192.168.0.10:5002 |
| Search Engine   | /search       | 192.168.0.10:5003 |
| System Settings | /settings     | 192.168.0.10:5004 |
+-----------------+---------------+-------------------+
```

The redirection works but it fails to load resources except for the index route (`/`), which is the `Main` server. If I put all the resources for each web server into the `Main` server project, then the redirection for each of the web servers, successfully loads the resources, but thats no ideal for us. How can I configure YARP to load the resources correctly?

**Comment by Tratcher** [comment, marked as answer]
Yes, YARP should be able to help with that.
Start with this guide: https://microsoft.github.io/reverse-proxy/articles/getting_started.html#configuration
Each IP would be its own cluster and destination, and each sub-directory would be a route. Something like this:
```JSON
"ReverseProxy": {
   "Routes": {
     "settingsroute" : {
       "ClusterId": "settingscluster",
       "Match": {
         "Path": "/settings/{**catch-all}"
       },
     }
   },
   "Clusters": {
     "settingscluster": {
       "Destinations": {
         "settingsdestination1": {
           "Address": "http://192.168.0.10:5004"
         }
       }
     }
   }
 }
```

**Reply by ibrahim-cmsb** [reply]
Hi Tratcher, I applied your solution but I still get the same issue, resource not loading.


### Redirect Page

![Redirect](https://user-images.githubusercontent.com/56328061/117239063-a7c91a00-ae60-11eb-802f-8c5eca470fea.PNG)

### Original Page

![Original](https://user-images.githubusercontent.com/56328061/117239335-2756e900-ae61-11eb-907e-d1ebce39c2a9.PNG)

### Config

```
"ReverseProxy": {
    "Routes": {
      "settingsroute": {
        "ClusterId": "settingscluster",
        "Match": {
          "Path": "/Account/{**catch-all}"
        }
      }
    },
    "Clusters": {
      "settingscluster": {
        "Destinations": {
          "settingsdestination1": {
            "Address": "http://192.168.0.61:97/"
          }
        }
      }
    }
  }
```

**Reply by Tratcher** [reply]
Oh I see, those resources aren't following the same routing rules. That's more of an issue with the HTML generation, it generates links for "/js" instead of "/Account/Js". That's not some thing the proxy can fix for you, it needs to be fixed in the back end site.

**Reply by ibrahim-cmsb** [reply]
Yes, I will have to edit those resources link according to routing rules. Thank you Tratcher.
